internal/proxy/github.go

```diff
@@ -53,13 +53,14 @@ func (p *Proxy) handleGitHubRequest(w http.ResponseWriter, r *http.Request, requ
 	requestLogEntry.WithField("json", gh).Debug("Processing GitHub JSON payload")
 
 	namespace, err := p.getUserWithRetry(gh.Repository.CloneURL, requestLogEntry, defaultRetry)
-	ns = namespace.Name
-	clusterURL := namespace.ClusterURL
-	requestLogEntry.WithFields(log.Fields{"ns": ns, "cluster": clusterURL, "repository": gh.Repository.CloneURL}).Info("Processing GitHub request ")
 	if err != nil {
 		p.HandleError(w, err, requestLogEntry)
 		return
 	}
+	ns = namespace.Name
+	clusterURL := namespace.ClusterURL
+	requestLogEntry.WithFields(log.Fields{"ns": ns, "cluster": clusterURL, "repository": gh.Repository.CloneURL}).Info("Processing GitHub request ")
```

Comment:
since you have the details of the ns and clusterURL here, it better to create a new logger with those fields added and use that to log all subsequent messages. 
e.g. 
```go 
nsLogger := requestLogEntry.WithFields(...)

# else where
nsLogger.Infof(...)
```

Reply:
ns and clusterURL are being logged only once. Why should I create a separate logger just for that?

Reply:
> ns and clusterURL are being logged only once. 
Actually no, look below for 
```go
log.WithField("ns", ns).
```

This information is useful while debugging as the inspecting the namespace usually leads to more clues.